### PR TITLE
[TIMOB-23327] iOS:  Introducing onStopBlackListUrl

### PR DESCRIPTION
**Pull request by AngelkPetkov**
[JIRA](https://jira.appcelerator.org/browse/TIMOB-23327?filter=-1)

### CODE REVIEW

File: apidoc/Titanium/UI/WebView.yml [hunk 1]
```diff
@@ -452,6 +452,13 @@ properties:
     default: false
     platforms: [iphone, ipad]
 
+  - name: blacklistFirstLoad
+    summary: Determines if the initial webview url will be checked against the blacklist.
+    platforms: [iphone,iphone]
+    type: Boolean
+    default: false
+    since: "5.4.0"
```
Comment (hansemannn) [inline apidoc/Titanium/UI/WebView.yml]: As discussed, I don't like that properly. As it also causes parity problems, we should remove it and check every URL that is in the blacklist. If they want to exclude certain domains, they should remove them from the blacklist or set the `blacklistedURLs` after its creation, which should be possible.

File: iphone/Classes/TiUIWebView.m [hunk 2]
```diff
@@ -385,6 +386,38 @@ -(BOOL)canGoForward
 	return [webview canGoForward];
 }
 
+-(void)setBlacklistedURLs_:(id)value
+{
+    if (blacklistedURLs == nil) {
+        blacklistedURLs = [[[NSMutableArray alloc] init] retain];
+    }
```
Comment (hansemannn) [inline iphone/Classes/TiUIWebView.m]: We need to release and re-create the array every-time we call this function (as discussed), so make sure it gets releases properly and set it again.

File: iphone/Classes/TiUIWebView.m [hunk 3]
```diff
@@ -672,21 +698,33 @@ -(CGFloat)contentWidthForWidth:(CGFloat)value
...
-		NSDictionary *event = newUrl == nil ? nil : [NSDictionary dictionaryWithObjectsAndKeys:[newUrl absoluteString], @"url", NUMINT(navigationType), @"navigationType", nil];
-		[self.proxy fireEvent:@"beforeload" withObject:event];
-	}
-
-	if (navigationType != UIWebViewNavigationTypeOther) {
-		RELEASE_TO_NIL(lastValidLoad);
-	}
-
+    
+    if ([self.proxy _hasListeners:@"beforeload"])
```
Comment (hansemannn) [inline iphone/Classes/TiUIWebView.m]: Indentation is messed up through the whole new code-block.

File: iphone/Classes/TiUIWebView.m [hunk 4]
```diff
@@ -672,21 +698,33 @@ -(CGFloat)contentWidthForWidth:(CGFloat)value
...
+    if (navigationType != UIWebViewNavigationTypeOther) {
+        RELEASE_TO_NIL(lastValidLoad);
+    }
+    
+    if (blacklistedURLs) {
+        for (NSString*check in blacklistedURLs)
+        {
+            if ([[newUrl absoluteString] rangeOfString:check options:NSCaseInsensitiveSearch].location != NSNotFound) {
+                if([self.proxy _hasListeners:@"onStopBlacklistedUrl"]) {
+                    NSDictionary* evt = [NSDictionary dictionaryWithObjectsAndKeys:[TiUtils stringValue:request], @"url", @"Webview did not load blacklisted url.", @"messsage", nil];
```
Comment (hansemannn) [inline iphone/Classes/TiUIWebView.m]: Did you test that you can cast a `request` of the type `NSURLRequest` to a string?

Comment (AngelkPetkov) [inline iphone/Classes/TiUIWebView.m]: Yeah i tested it to see if it works, updated the demo-code in the ticket as well with the use-case i used. Also newURL is an NSURL not an NSURLRequest

Comment (hansemannn) [inline iphone/Classes/TiUIWebView.m]: Check it again:
```objc
NSDictionary* evt = [NSDictionary dictionaryWithObjectsAndKeys:[TiUtils stringValue:request], @"url", @"Webview did not load blacklisted url.", @"messsage", nil];
```

uses the variable `request` which is no `NSURL` (which should also be received correctly).

Comment (AngelkPetkov) [inline iphone/Classes/TiUIWebView.m]: Oh that sorry, thought you was talking about the if statement. Yeah it works, i cant pass in the URL as it causes a weird bug which we disguised, where the page will finish loading before reaching the return statement.

File: iphone/Classes/TiUIWebView.m [hunk 5]
```diff
@@ -672,21 +698,33 @@ -(CGFloat)contentWidthForWidth:(CGFloat)value
...
 	if (navigationType != UIWebViewNavigationTypeOther) {
 		RELEASE_TO_NIL(lastValidLoad);
 	}
-
+    
+	if (blacklistedURLs) {
+           for (NSString*check in blacklistedURLs)
+           {
+               if ([[newUrl absoluteString] rangeOfString:check options:NSCaseInsensitiveSearch].location != NSNotFound) {
+                   if([self.proxy _hasListeners:@"onStopBlacklistedUrl"]) {
```
Comment (yomybaby) [inline iphone/Classes/TiUIWebView.m]: This event name should be `stopBlacklistedUrl` without `on`, right?

**Pull request closed without merging**